stripe/src/main/java/com/stripe/android/Stripe.java

```diff
@@ -133,25 +149,36 @@ public void createToken(
             @NonNull @Size(min = 1) final String publishableKey,
             @Nullable final Executor executor,
             @NonNull final TokenCallback callback) {
-        try {
-            if (card == null) {
-                throw new RuntimeException(
-                        "Required Parameter: 'card' is required to create a token");
-            }
+        if (card == null) {
+            throw new RuntimeException(
+                    "Required Parameter: 'card' is required to create a token");
+        }
 
-            if (callback == null) {
-                throw new RuntimeException(
-                        "Required Parameter: 'callback' is required to use the created " +
-                                "token and handle errors");
-            }
+        createToken(hashMapFromCard(card), publishableKey, executor, callback);
+    }
 
-            validateKey(publishableKey);
+    /**
+     * Call to create a {@link Token} for a {@link BankAccount} with the publishable key and
+     * {@link Executor} specified.
+     *
+     * @param bankAccount the {@link BankAccount} for which to create a {@link Token}
+     * @param publishableKey the publishable key to use
+     * @param executor an {@link Executor} to run this operation on. If null, this is run on a
+     *                 default non-ui executor
+     * @param callback a {@link TokenCallback} to receive the result or error message
+     */
+    public void createToken(
+            @NonNull final BankAccount bankAccount,
+            @NonNull @Size(min = 1) final String publishableKey,
+            @Nullable final Executor executor,
+            @NonNull final TokenCallback callback) {
 
-            tokenCreator.create(card, publishableKey, executor, callback);
-        }
-        catch (AuthenticationException e) {
-            callback.onError(e);
+        if (bankAccount == null) {
+            throw new RuntimeException(
+                    "Required parameter: 'bankAccount' is requred to create a token");
         }
+
+        createToken(hashMapFromBankAccount(bankAccount), publishableKey, executor, callback);
```

Comment:
This could seem recursive. The naming change suggested above might help deciphering the different currently overloaded methods.